beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -30,90 +30,82 @@ pub enum BlobError {
     ///
     /// We were unable to process this sync committee message due to an internal error. It's unclear if the
     /// sync committee message is valid.
     BeaconChainError(BeaconChainError),
-    /// No blobs for the specified block where we would expect blobs.
-    UnavailableBlobs,
-    /// Blobs provided for a pre-Deneb fork.
-    InconsistentFork,
 
-    /// The `blobs_sidecar.message.beacon_block_root` block is unknown.
+    /// The `BlobSidecar` was gossiped over an incorrect subnet.
     ///
     /// ## Peer scoring
     ///
-    /// The blob points to a block we have not yet imported. The blob cannot be imported
-    /// into fork choice yet
-    UnknownHeadBlock {
-        beacon_block_root: Hash256,
-    },
-
-    /// The `BlobSidecar` was gossiped over an incorrect subnet.
-    InvalidSubnet {
-        expected: u64,
-        received: u64,
-    },
+    /// The blob is invalid or the peer is faulty.
+    InvalidSubnet { expected: u64, received: u64 },
 
     /// The sidecar corresponds to a slot older than the finalized head slot.
+    ///
+    /// ## Peer scoring
+    ///
+    /// It's unclear if this blob is valid, but this blob is for a finalized slot and is
+    /// therefore useless to us.
     PastFinalizedSlot {
         blob_slot: Slot,
         finalized_slot: Slot,
     },
 
     /// The proposer index specified in the sidecar does not match the locally computed
     /// proposer index.
-    ProposerIndexMismatch {
-        sidecar: usize,
-        local: usize,
-    },
+    ///
+    /// ## Peer scoring
+    ///
+    /// The blob is invalid and the peer is faulty.
+    ProposerIndexMismatch { sidecar: usize, local: usize },
 
+    /// The proposal signature in invalid.
+    ///
+    /// ## Peer scoring
+    ///
+    /// The blob is invalid and the peer is faulty.
     ProposerSignatureInvalid,
 
-    /// A sidecar with same slot, beacon_block_root and proposer_index but different blob is received for
-    /// the same blob index.
-    RepeatSidecar {
-        proposer: usize,
-        slot: Slot,
-        blob_index: usize,
-    },
-
     /// The proposal_index corresponding to blob.beacon_block_root is not known.
     ///
     /// ## Peer scoring
     ///
     /// The block is invalid and the peer is faulty.
     UnknownValidator(u64),
 
-    BlobCacheError(AvailabilityCheckError),
+    /// The provided blob is not from a later slot than its parent.
+    ///
+    /// ## Peer scoring
+    ///
+    /// The blob is invalid and the peer is faulty.
+    BlobIsNotLaterThanParent { blob_slot: Slot, parent_slot: Slot },
+
+    /// The provided blob's parent block is unknown.
+    ///
+    /// ## Peer scoring
+    ///
+    /// We cannot process the blob without validating its parent, the peer isn't necessarily faulty.
+    BlobParentUnknown {
+        blob_root: Hash256,
+        blob_parent_root: Hash256,
+    },
+
+    /// The given blob has already been seen for the given `(sidecar.block_root, sidecar.index)` tuple
```

Comment:
This blob might not be the same right? it's simply the blob identifier that has been already seen
```suggestion
    /// A blob has already been seen for the given `(sidecar.block_root, sidecar.index)` tuple
```